spec/generators/rails-template.rb

```diff
@@ -0,0 +1,269 @@
+      def change
+        create_table :wizards do |t|
+          t.string :name
+          t.integer :house
+          t.string :parent_email
+          t.text :notes
+    
+          t.timestamps
+        end
+      end
+    end
+  RUBY
+
+  file "db/migrate/20190620000002_create_wands.rb", <<~RUBY
+    class CreateWands < #{migration_superclass}
+      def change
+        create_table :wands do |t|
+          t.references :wizard, unique: true, null: false
+          t.string :wood_type
+          t.integer :core_type
+    
+          t.timestamps
+        end
+      end
+    end
+  RUBY
+
+  file "db/migrate/20190620000003_create_spell_books.rb", <<~RUBY
+    class CreateSpellBooks < #{migration_superclass}
+      def change
+        create_table :spell_books do |t|
+          t.string :name
+          t.references :wizard
+        end
+      end
+    end
+  RUBY
+
+  file "db/migrate/20190620000004_add_more_column_types_to_wands.rb", <<~RUBY
+    class AddMoreColumnTypesToWands < #{migration_superclass}
+      def change
+        add_column :wands, :flexibility,    :float,   null: false, default: 0.5
+        add_column :wands, :hardness,       :decimal, null: false, precision: 10, scale: 10, default: 5
+        add_column :wands, :reflectance,    :decimal, null: false, precision: 10, scale: 0, default: 0.5
+        add_column :wands, :broken,         :boolean, null: false, default: false
+        add_column :wands, :chosen_at_date, :date
+        add_column :wands, :chosen_at_time, :time
+        # JSON column type is only supported on 5.2 or higher
+        unless ['4.2', '5.0', '5.1'].include?(ENV['RAILS_VERSION'])
+          add_column :wands, :spell_history,  :json
+          add_column :wands, :maker_info,     :json,    null: false, default: '{}'
+        end
+      end
+    end
+  RUBY
+end
+
+def add_sorbet_test_files
+  file "typed-override.yaml", <<~YAML
+    true:
+    - ./sorbet_test_cases.rb
+  YAML
+
+  copy_file "./sorbet_test_cases.rb", "sorbet_test_cases.rb"
+end
+
+# Get files relative to this template when copying.
+def source_paths
+  [__dir__]
+end
+
+# Main setup
+source_paths
+
+if ['4.2', '5.0'].include?(ENV["RAILS_VERSION"])
+  File.open('Gemfile', 'r+') do |f|   
+    out = ""
+    f.each do |line|
+      # We remove sdoc and web-console because they misbehave.
+      # sqlite needs to be limited to 1.3.x or it won't work.
+      out << line.gsub("gem 'sqlite3'", "gem 'sqlite3', '~> 1.3.6'") unless line =~ /gem \'(sdoc|web-console)\'.*/
+    end
+    f.pos = 0
+    f.print out
+    f.truncate(f.pos)
+  end
+end
+
+add_gems
+
+after_bundle do
+  say "Creating application..."
+  add_routes
+  create_initializers
+  create_lib
+  create_helpers
+  create_models
+  create_migrations
+  add_sorbet_test_files
+  
+  # Rails 4.2 doesn't have the rails_command method.
+  if ['4.2'].include?(ENV["RAILS_VERSION"])
+    run "bundle exec rake db:migrate"
+  else
+    rails_command "db:migrate"
```

Comment:
if you want to simplify it, we can always use `bundle exec rake db:migrate`. It works on all versions of Rails.